.github/workflows/deployColl.yml

```diff
@@ -0,0 +1,87 @@
+on:
+  release:
+    types: [created]
+    tags:
+      - 'coll-v*\.*\.*'
+
+name: AWS PCM Build and Deploy Collaudo
+
+jobs:
+  deploy:
+    if: ${{ startsWith(github.ref, 'refs/tags/coll') }}
+    name: Deploy
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Print GITHUB_REF
+      run: echo "GITHUB_REF=${GITHUB_REF}"
+    - name: Set output
+      id: vars
+      run: echo "tag=${GITHUB_REF#refs/*/}" | sed 's/coll-v//' >> "${GITHUB_OUTPUT}"
+    - name: Check output
+      env:
+        RELEASE_VERSION: ${{ steps.vars.outputs.tag }}
+      run: |
+        echo $RELEASE_VERSION
+        echo ${{ steps.vars.outputs.tag }}
+
+    - name: Configure AWS credentials
+      uses: aws-actions/configure-aws-credentials@v1
+      with:
+        aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        aws-region: eu-south-1
+
+    - name: Login to Amazon ECR
+      id: login-ecr
+      uses: aws-actions/amazon-ecr-login@v1
+
+    - name: Build, tag, and push the image to Amazon ECR
+      id: build-image
+      env:
+        ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+        ECR_REPOSITORY: ${{ secrets.REPO_NAME }}
+        IMAGE_TAG: ${{ steps.vars.outputs.tag }}
+      run: |
+        # Build a docker container and push it to ECR 
+        docker build --build-arg GEOIP_LICENSE=${{ secrets.GEOIP_LICENSE }} -t $ECR_REGISTRY/$ECR_REPOSITORY:$IMAGE_TAG .
+        echo "Pushing image to COLL ECR..."
+        docker push $ECR_REGISTRY/$ECR_REPOSITORY:$IMAGE_TAG
+        echo "::set-output name=image::$ECR_REGISTRY/$ECR_REPOSITORY:$IMAGE_TAG"
+        echo "Pushed image to COLL ECR "
+
+    - name: Update kube config for coll EKS
+      id: update-kube-config-coll
+      run: aws eks update-kubeconfig --name ${{ secrets.EKS_CLUSTER_NAME }}
+
+    - name: Apply the deployment to EKS
+      id: deploy
+      env:
+        IMAGE_TAG: ${{ steps.build-image.outputs.image }}
+        ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+        ECR_REPOSITORY: ${{ secrets.REPO_NAME }}
+      run: |
+        echo "Image tag: $IMAGE_TAG"
+        tag=$(echo ${IMAGE_TAG##*:})
+        echo "coll ecr: $ECR_REGISTRY/$ECR_REPOSITORY:$tag"
+        echo "Deploying to test EKS..." 
+        cat git-deployment-coll.yml | sed "s|ImagePlaceholder|$ECR_REGISTRY/$ECR_REPOSITORY:$tag|g" | kubectl apply -f -
+
+    - name: Check the deploy to EKS COLL
+      id: deploy-eks-coll
+      run: |
+        kubectl get cronjobs -n dtd-crawler-coll
```

Comment:
`kubectl get cronjobs` returns zero even in the case when no cronjob resource is present.

Reply:
Hi @tensor5,

The actions are already being run on a github actions runner, so not on our AWS infrastructure
(https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners) I used a ubuntu runner since that is perfectly suited for our needs to just build the image.

I wasn't aware that you had a github registry also, I know that the free tier offers a very low amount of space (https://github.com/features/packages#pricing) currently the latest image is around 1.2GB, we would have to see how much storage there is in your github registry if we want to move the image there, it's doable very quickly so that will not be an issue we just need to decide if that is what suits you best .

As for the 

> kubectl get cronjobs returns zero even in the case when no cronjob resource is present.

That's right, because the rc of the bash command in this case is 0 even if there is no cronjob present, when testing I tried some checks and with this I was just simply checking to see that they were updated, I don't know why but at the time I didn't think of doing it this way:

`for cronjob in $(kubectl get cronjobs | awk '{print $1}' | grep -iv name); do kubectl get cronjob ${cronjob} -o json | jq -r '.spec.jobTemplate.spec.template.spec.containers[].image' | cut -d ':' -f2 | while read result; do { [[ ${result} == ${tag} ]] && echo "Deployment ${cronjob} ok"; } || { echo "Deployment ${cronjob} ko" && exit 1; }; done ; done`

It's a bit of a mess to read, but what I do is get the list of the cronjobs and then iterate over each one of them and check that the image tag they are using is the same one as the new version tag and if it's not then I do an exit 1 thus forcing a rc of 1 and making the task fail

Here is how it would work:

>[ec2-user@ip-172-17-1-242 ~]$ tag=2.2.1
`[ec2-user@ip-172-17-1-242 ~]$ for cronjob in $(kubectl get cronjobs | awk '{print $1}' | grep -iv name); do kubectl get cronjob ${cronjob} -o json | jq -r '.spec.jobTemplate.spec.template.spec.containers[].image' | cut -d ':' -f2 | while read result; do { [[ ${result} == ${tag} ]] && echo "Deployment ${cronjob} ok"; } || { echo "Deployment ${cronjob} ko" && exit 1; }; done ; done`
Deployment dtd-crawler-pa2026-manager-job-prod ko
Deployment dtd-crawler-queue-manager-job-prod ko
Deployment dtd-crawler-scan-manager-job-prod ko
>[ec2-user@ip-172-17-1-242 ~]$ echo $?
>1
>[ec2-user@ip-172-17-1-242 ~]$
>[ec2-user@ip-172-17-1-242 ~]$
>[ec2-user@ip-172-17-1-242 ~]$ tag=2.2.2
`[ec2-user@ip-172-17-1-242 ~]$ for cronjob in $(kubectl get cronjobs | awk '{print $1}' | grep -iv name); do kubectl get cronjob ${cronjob} -o json | jq -r '.spec.jobTemplate.spec.template.spec.containers[].image' | cut -d ':' -f2 | while read result; do { [[ ${result} == ${tag} ]] && echo "Deployment ${cronjob} ok"; } || { echo "Deployment ${cronjob} ko" && exit 1; }; done ; done`
>Deployment dtd-crawler-pa2026-manager-job-prod ok
>Deployment dtd-crawler-queue-manager-job-prod ok
>Deployment dtd-crawler-scan-manager-job-prod ok
>[ec2-user@ip-172-17-1-242 ~]$ echo $?
>0

Now, sadly this doesn't fix the issue of having the rc 0 on kubectl get <kind> I tried in a few ways to get around it and found some issues related to similar issues where the rc is 0 when it would be useful to have a different one instead with no concrete answer but I think I managed to come up with something that could help us understand if there are no cronjob objects

` kubectl get cronjobs -o json | jq '.items' -r | wc -l | while read result; do [[ ${result} != 1 ]] && exit 0; done` 

I basically just check how many lines I get back from running ` kubectl get cronjobs -o json | jq '.items' -r`, because if there are no such objects then the result will be:
>[ec2-user@ip-172-17-1-242 ~]$ kubectl get cronjobs -o json | jq '.items' -r
>[]

So if the result is not 1 (so an empty list of items) then we should be ok.

Let me know if you think this sort of check (the one checking the image tag on the cronjobs is the same as the release tag that triggered the action and the one on the cronjob rc, I think it would make sense to firstly check that there are cronjobs present and then check their tag version) could work and I will be more than happy to change it

Thank you!